packages/api-docs-builder/ApiBuilders/ComponentApiBuilder.ts

```diff
@@ -735,14 +735,19 @@ export default async function generateComponentApi(
   let reactApi: ComponentReactApi;
 
   try {
-    reactApi = docgenParse(src, null, defaultHandlers.concat(muiDefaultPropsHandler), {
-      filename,
-    });
+    reactApi = docgenParse(
+      src.replace(/prop-types-compat/gm, 'prop-types'),
```

Comment:
This is required to trick react-docgen to produce the same result, otherwise it will think that the PropTypes from `prop-types-compat` are all custom prop types.